python/revng/cli/commands_registry.py

```diff
@@ -82,7 +83,12 @@ def register_external_command(self, command: str, path: str):
         self.register_command(ExternalCommand(self._parse_command(command), path))
 
     def run(self, arguments, options: Options):
-        (args, rest) = self._create_parser().parse_known_args(arguments)
+        parser = self._create_parser()
+        with contextlib.suppress(Exception):
+            import argcomplete
+
+            argcomplete.autocomplete(parser)
+        (args, rest) = parser.parse_known_args(arguments)
```

Comment:
I'm a bit hesitant, main reasons:
* argcomplete requires adding a command to `.${SHELL}rc` to work: https://kislyuk.github.io/argcomplete/#activating-global-completion
* right now it's quite limited since most of the commands are passthroughs to C++ executables

Reply:
Uhm, OK, then it doesn't look like mergeable. We probably need more time to look into a full proper solution.

Reply:
Stashed it for later, dropped in the current PR